src/main/typescript/components/App.tsx

```diff
@@ -1,47 +1,103 @@
 import * as React from "react"
+import { BrowserRouter, Switch, Route, Link, useParams } from "react-router-dom"
 import styled, { createGlobalStyle } from "styled-components"
 
 import AddButton from './AddButton'
 import * as AppReducer from "./AppReducer"
 import { DeletedGamesTable } from "./DeletedGamesTable"
 import { GameTable } from "./GameTable"
+import { GroupSelectorPage } from "./GroupSelectorPage"
 import { NewGameForm } from "./NewGameForm"
 import { RatingTable } from "./RatingTable"
 
 export function App() {
   const [state, dispatch] = React.useReducer(AppReducer.reducer, AppReducer.initialState)
-  const { group, showAddForm } = state
 
   React.useEffect(
-    () => {
-      AppReducer.fetchGames(dispatch, group.id)
-      AppReducer.fetchDeletedGames(dispatch, group.id)
+    () => AppReducer.fetchGroups(dispatch),
+    []
+  )
+
+  return <BrowserRouter>
+    <GlobalStyle />
+    <Container>
+      <Switch>
+        <Route exact path="/">
+          <GroupSelectorPage groups={state.groups} />
+        </Route>
+        <Route exact path="/:groupUrlName">
+          <GroupPage dispatch={dispatch} state={state} />
+        </Route>
+      </Switch>
+    </Container>
+  </BrowserRouter>
+}
+
+interface GroupPageProps {
+  dispatch: AppReducer.Dispatch,
+  state: AppReducer.State
+}
+
+function GroupPage({ dispatch, state }: GroupPageProps) {
+  const { groups, openGroup, showAddForm } = state
+
+  const { groupUrlName } = useParams()
+
+  React.useEffect(
+    function closeGroupOnUnmount() {
```

Comment:
the callback of the useEffect can be more simple, like 

`
 React.useEffect(() => {
        dispatch({ type: "groupClosed" })
        AppReducer.fetchGroups(dispatch)
    },
    []
  )`

Reply:
I used a named function to make the intent clearer, but it arguably doesn't do much for a 2-line function.